src/meta/client/MetaClient.cpp

```diff
@@ -1779,5 +1779,38 @@ Status MetaClient::refreshCache() {
     return ret ? Status::OK() : Status::Error("Load data failed");
 }
 
+StatusOr<LeaderMap> MetaClient::loadLeader() {
+    // Return error if has not loadData before
+    if (!ready_) {
+        return Status::Error("Not ready!");
+    }
+
+    auto ret = listHosts().get();
+    if (!ret.ok()) {
+        return Status::Error("List hosts failed");
+    }
+
+    LeaderMap leaderMap;
+    auto hostItems = std::move(ret).value();
+    for (auto& item : hostItems) {
+        auto hostAddr = HostAddr(item.hostAddr.ip, item.hostAddr.port);
+        for (auto& spaceEntry : item.get_leader_parts()) {
+            auto spaceName = spaceEntry.first;
+            auto status = getSpaceIdByNameFromCache(spaceName);
+            if (!status.ok()) {
+                continue;
```

Comment:
Why continue here?

Reply:
Cause `show hosts` return map of <space name, leader>, we have to convert to spaceId here. We just can ignore the if the space exists in `show hosts` but not in `show spaces`, actually they are both from meta, the `continue` rarely can hit.

Reply:
Fine.